pandas/core/construction.py

```diff
@@ -412,6 +412,9 @@ def sanitize_array(
 
     # GH#846
     if isinstance(data, np.ndarray):
+        if copy is None:
+            # copy by default for DataFrame({"A": ndarray})
+            copy = True
```

Comment:
so you are changing the default here? this certainly needs a deprecation

Reply:
https://github.com/pandas-dev/pandas/pull/34872#issuecomment-646850479 has the best summary. There's no API-breaking changes here.

Reply:
ok, umm why aren't we not copying ndarrays though? I mean the default is false, happy to take a view for ndarrays' but this seems strange to do opposite things.

Reply:
I'm not sure. I don't think we should change it here.

If we want to deprecate the default behavior for ndarray or EA then that can be discussed as a followup. It should be relatively simple compared to this PR.

Reply:
I do not think we should do this, it is a horribly divergent behavior. We are deprecating tiny little things, this is a major change and am -1 unless this has the same behaviror for EA and ndarrays.

Reply:
We can't make API breaking changes in a minor release. And I don't see a cleaner way to solve https://github.com/pandas-dev/pandas/issues/32960 than making `copy` apply to dict inputs too.

I'm fine with aligning the default behavior between ndarray and EA, but that will have to wait for 2.0.

Reply:
Although, now that I see your objection, other uses of `sanitize_array` will need to be checked. I've only updated `DataFrame.__init__`, but I'll need to ensure other places use `copy=False` if they were relying on the default.